### PR TITLE
Randomize instance order and crop

### PR DESCRIPTION
**Pull request by aaprasad**
As noted in #31, the model may be overfitting to the order in which the instance reid_features are fed into the model. Especially with our dataset logic in microscopy where we just iterate over the track ids this could be especially problematic. Thus in this PR we:
* shuffle instances in training mode.

We also add a functionality for random anchor selection when we crop to test if we can generalize the model to missing anchors

## Summary by CodeRabbit

- **New Features**
  - Introduced new attributes for managing centroid and anchor data, enhancing data structuring capabilities.
  - Implemented conditional data augmentation and shuffling for training mode in various datasets to optimize model training.
  
- **Enhancements**
  - Improved bounding box calculation using a more accurate centering technique.
  
- **Bug Fixes**
  - Standardized `mode` attribute handling by converting it to lowercase across multiple dataset classes to ensure consistent behavior.

### CODE REVIEW

File: biogtr/data_structures.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def __init__(
         pred_track_id: int = -1,
         bbox: ArrayLike = torch.empty((0, 4)),
         crop: ArrayLike = torch.tensor([]),
+        centroid: dict[str, ArrayLike] = None,
```
[the remark this answers is not in the document]
Comment (talmo) [reply biogtr/data_structures.py:19]: Bad! As is initializing with the `torch.empty` and `torch.tensor` above since these are all mutable!

Just change the type to `centroid: Optional[dict[str, ArrayLike]] = None` or `centroid: dict[str, ArrayLike] | None = None` (newer python syntax)

**Review by talmo — APPROVED** [review]

File: biogtr/data_structures.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaprasad) [reply biogtr/data_structures.py:19]: I opened an issue to go through and remove all the mutable defaults.
